robots/robotPositioner.h

```diff
@@ -0,0 +1,181 @@
+            m_k1(k1),
+            m_k2(k2),
+            m_alpha(alpha),
+            m_beta(beta),
+            m_max_velocity(max_velocity) 
+    {  }
+
+    //! sets the goal pose (x, y, angle)
+    void setGoalPose(millimeter_t pos_x, millimeter_t pos_y, degree_t goal_angle) 
+    {
+        m_goalPositionX = pos_x;
+        m_goalPositionY = pos_y;
+        m_goalAngle = goal_angle;
+    }
+
+
+    //! updates the velocities in order to get to a goal location. This function can be used
+    //! without a robot interface, where only velocities are calculated but no robot actions
+    //! will be executed.
+    void updateVelocities(
+        const millimeter_t pos_x,         // current X position of robot
+        const millimeter_t pos_y,         // current Y position of robot
+        const degree_t angle,             // current heading of robot
+        meters_per_second_t &v,           // velocity to update
+        degrees_per_second_t &w)          // angular velocity to update
+
+    {        
+
+        m_pos_X = pos_x;
+        m_pos_Y = pos_y;
+        m_heading = angle;
+        
+        // updates the range and bearing
+        updateRangeAndBearing();
+
+        // orientation of Target with respect to the line of sight from the observer to the target
+        degree_t theta = m_heading + m_bearingFromGoal - m_goalAngle; 
+        theta = angleWrapAround(theta);
+
+
+        auto k = -(((m_k2 * (m_bearingFromGoal - radian_t(atan( radian_t(-m_k1 * theta).value()) ))))
+         + degree_t( 1+(m_k1/(1+pow(m_k1*theta.value(),2))) )
+         * units::math::sin(m_bearingFromGoal))/m_distanceFromGoal;
+     
+        v = m_max_velocity/scalar_t( (1+m_beta*pow(abs(k.value()),m_alpha)));
+        w = k*v;  
+    }
+
+    
+    //! This function will update the motors so it drives towards a previously set goal location
+    void updateMotors(BoBRobotics::Robots::Tank &bot, 
+                      const millimeter_t pos_x,
+                      const millimeter_t pos_y,       
+                      const degree_t angle)
+    {           
+        // here we are solving simultaneous equations where we would like to
+        // recover Vl (left wheel velocity) and Vr (right wheel velocity) 
+        //                                       v = wheel_radius * (Vl+Vr)/2 
+        //                                       w = wheel_radius * (Vr-Vl)/axis_length
+
+        meters_per_second_t v;
+        degrees_per_second_t w; 
+        updateVelocities(pos_x, pos_y, angle, v, w); 
+
+        const millimeter_t robot_wheel_radius = bot.getRobotWheelRadius();
+        const millimeter_t robot_axis_length  = bot.getRobotAxisLength();
+        
+        const double a = (robot_wheel_radius/2).value(); 
+        const double b = (robot_wheel_radius/robot_axis_length).value();
+    
+        auto mmps = 1_mm/1_s;
+        mmps = v; // converting to mm/s
+        const double c = v.value();
```

Comment:
I'm guessing this should read:
``
const double c = mmps.value();
``
...otherwise ``c`` will still be in m/s, rather than mm/s (which I guess is what you intended?).